apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs

```diff
@@ -96,6 +95,14 @@ IEnumerator<Row> MergeResponseChunks(ReadRowsResponse response)
                 {
                     if (chunk.ResetRow)
                     {
+                        Assert(
```

Comment:
I realise `Assert` is already in the code; but why aren't we using `GaxPreconditions...` instead?

Reply:
We need to perform additional work on failure: reset the row merger's state.